From 5ea48269b36e118423f6c4f40b34d1cd0928919a Mon Sep 17 00:00:00 2001
From: Lei Maohui <leimaohui@fujitsu.com>
Date: Thu, 13 Jun 2024 02:01:51 +0000
Subject: [PATCH] FIx build errors with gcc14.

jdk8u-shenandoah-aarch64-shenandoah-jdk8u272-b10/jdk/src/share/native/sun/misc/URLClassPath.c:58:5: error: implicit declaration of function 'VerifyFixClassname' [-Wimplicit-function-declaration]
   58 |     VerifyFixClassname(clname);
      |     ^~~~~~~~~~~~~~~~~~

jdk8u-shenandoah-aarch64-shenandoah-jdk8u272-b10/jdk/src/share/native/sun/misc/URLClassPath.c:60:10: error: implicit declaration of function 'VerifyClassname' [-Wimplicit-function-declaration]
   60 |     if (!VerifyClassname(clname, JNI_TRUE)) {  /* expects slashed name */
      |          ^~~~~~~~~~~~~~~
openjdk-8/272/jdk8u-shenandoah-aarch64-shenandoah-jdk8u272-b10/jdk/src/share/native/sun/misc/URLClassPath.c:68:9: error: implicit declaration of function 'free' [-Wimplicit-function-declaration]
   68 |         free(clname);
      |         ^~~~

Upstream-Status: Pending
Signed-off-by: Lei Maohui <leimaohui@fujitsu.com>
---
 jdk/src/share/native/sun/misc/URLClassPath.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/jdk/src/share/native/sun/misc/URLClassPath.c b/jdk/src/share/native/sun/misc/URLClassPath.c
index e0ad4f81..34040959 100644
--- a/jdk/src/share/native/sun/misc/URLClassPath.c
+++ b/jdk/src/share/native/sun/misc/URLClassPath.c
@@ -24,7 +24,7 @@
  */
 
 #include <string.h>
-
+#include <stdlib.h>
 #include "jni.h"
 #include "jni_util.h"
 #include "jlong.h"
@@ -33,6 +33,8 @@
 
 #include "sun_misc_URLClassPath.h"
 
+extern jboolean VerifyFixClassname(char *utf_name);
+extern jboolean VerifyClassname(char *utf_name, jboolean arrayAllowed);
 extern char*
 getUTF(JNIEnv *env, jstring str, char* localBuf, int bufSize);
 
-- 
2.34.1

